modules/launcher/src/main/java/org/mule/module/launcher/util/ObservableList.java

```diff
@@ -323,14 +324,16 @@ public boolean hasNext()
         @Override
         public E next()
         {
-            cursor++;
-            return iterDelegate.next();
+            E next = iterDelegate.next();
+            lastRet = cursor++;
```

Comment:
could you elaborate a little bit on what was the root cause of the issue and why you consider that this lastRet is the best solution for the issue?